packages/css/src/plugins/at-rules/types.ts

```diff
@@ -0,0 +1,84 @@
+import type { AtRule, Rule } from 'postcss';
+
+export type Property = 'width' | 'height' | 'device-width' | 'device-height';
+export type ComparisonOperator = '<=' | '=' | '>=' | '<' | '>';
+export type LengthUnit = 'ch' | 'em' | 'ex' | 'px' | 'rem';
```

Comment:
I could be missing it, but if we see say `min-width: 32px` and `min-width: 2rem`, are these sorted as equals±?  Technically we can't make assumptions on some of these, so not even sure how we would.

Just not sure if it's said anywhere, we make assumptions about the sorting, but it's deterministic…I think at the end of the day, that's it, it's deterministic and should be fit for our purpose—could have some hard bugs as this code is a **_lot_** 😮 (good job), but should fix more bugs than it has!

Reply:
> Technically we can't make assumptions on some of these, so not even sure how we would. Might just be worth documenting, we only sort against same unit'd values and expect all media queries to be the same unit…? (if I'm understanding the code well)

oh yeah it would be pretty easy to normalise different units

do you anticipate any developers mixing units like this? i wasn't sure about how much value it would provide

Reply:
I do anticipate mixing `rem` and `px` as ADS ships `rem` and I see `px` in use a lot actually, I think even within our products or WAC you might find this dozens of times.

_Perhaps_ having some like `mediaQueryRemRatio: 16` defaulted/set somewhere to convert `em` and `rem` to `px` and normalize around `px` (configurable or not, I don't care) could work?

I can't tell you what `ch` or `ex` to `px` is though, and this might just be scope creeping, maybe worth testing what mixing `rem` and `px` looks like is sufficient…

Reply:
ahh right 👍  yeah i'll add it in

Reply:
This was added, LGTM: https://github.com/atlassian-labs/compiled/pull/1666/files#diff-e6c66e3c597ed6c80303321a9daac2eaabb0f46a431faea945997f8c4eb1a23bR145